video/out/hwdec/dmabuf_interop_wl.c

```diff
@@ -35,6 +32,32 @@ static bool map(struct ra_hwdec_mapper *mapper,
                 struct dmabuf_interop *dmabuf_interop,
                 bool probing)
 {
+    // 1. only validate format when composed layers is enabled (i.e. vaapi)
+    // 2. for drmprime, just return true for now, as this use case
+    // has not been tested.
+    if (!dmabuf_interop->composed_layers)
+        return true;
+
+    int layer_no = 0;
+    struct dmabuf_interop_priv *mapper_p = mapper->priv;
+    uint32_t drm_format = mapper_p->desc.layers[layer_no].format;
+
+    if (mapper_p->desc.nb_layers != 1) {
```

Comment:
Not sure if this is actually possible to be at zero or less, but would actually seem like a better way to first check, then utilize?  I see desc being an AVDRMFrameDescriptor which has a static amount of planes and formats always available so positively enough even without checking there should be no out of invalid memory access.

Not sure about the message level. Unless this is spammy, maybe follow other probing components and do `MP_MSG` with `probing ? MSGL_V : MSGL_ERR`?

```c
    // we're only reading things from this structure so const should be fine
    const AVDRMFrameDescriptor desc =
        ((struct dmabuf_interop_priv *)mapper->priv)->desc;

    if (desc.nb_layers != 1 || desc.nb_objects != 1) {
        MP_DBG(mapper, "Mapping attempted with surface which has unexpected "
                       "amount of layers (%s, %d) or objects (%s, %d)!\n",
               desc.nb_layers != 1 ? "yes" : "no",
               desc.nb_layers,
               desc.nb_objects != 1 ? "yes" : "no",
               desc.nb_objects);
        return false;
    }

    uint32_t drm_format      = desc.layers[0].format;
    uint64_t format_modifier = desc.objects[0].format_modifier;

    if (!ra_compatible_format(mapper->ra, drm_format, format_modifier) {
        MP_DBG(mapper, "Mapping attempted with an unsupported surface with "
                       "format %s; drm format '%s(%016lx)'!\n",
                       mp_imgfmt_to_name(mapper->src->params.hw_subfmt),
                       mp_tag_str(drm_format),
                       format_modifier);
        return false;
    }

    MP_DBG(mapper, "Mapping supported Wayland display format: '%s(%016lx)'\n",
           mp_tag_str(drm_format), format_modifier);
```

Reply:
Made it always verbose.